src/controllers/announcement_controller.cr

```diff
@@ -83,6 +83,11 @@ class AnnouncementController < ApplicationController
     end
   end
 
+  def random
+    announcement = Announcement.random
+    redirect_to "/announcements/#{announcement.id}"
+  end
```

Comment:
Just to avoid runtime error when db is empty. 
```crystal
    if announcement = Announcement.random
      redirect_to "/announcements/#{announcement.id}"
    else
      redirect_to "/"
    end

```